source/common/filesystem/kqueue/watcher_impl.cc

```diff
@@ -116,20 +117,28 @@ absl::Status WatcherImpl::onKqueueEvent() {
 
     absl::StatusOr<PathSplitResult> pathname_or_error =
         file_system_.splitPathFromFilename(file->file_);
-    RETURN_IF_NOT_OK_REF(pathname_or_error.status());
+    if (!pathname_or_error.ok()) {
```

Comment:
A pathname failure is not temporary, will continuing means you may warn forever on every event and never recover?